riposte-core/src/main/java/com/nike/riposte/server/handler/RequestInfoSetterHandler.java

```diff
@@ -47,13 +59,31 @@ else if (msg instanceof HttpContent) {
                                                     + "HttpProcessingState. This should be impossible");
                 }
 
-                requestInfo.addContentChunk((HttpContent) msg);
+                int currentRequestLengthInBytes = requestInfo.addContentChunk((HttpContent) msg);
+                int configuredMaxRequestSize = getConfiguredMaxRequestSize(state);
+                
+                if (!isMaxRequestSizeValidationDisabled(configuredMaxRequestSize) && currentRequestLengthInBytes > configuredMaxRequestSize) {
+                    throw new TooLongFrameException("Request raw content length exceeded configured max request size of " + configuredMaxRequestSize);
+                }
             }
         }
 
         return PipelineContinuationBehavior.CONTINUE;
     }
 
+    private boolean isMaxRequestSizeValidationDisabled(int configuredMaxRequestSize) {
+        return configuredMaxRequestSize <= 0;
+    }
+
+    private int getConfiguredMaxRequestSize(HttpProcessingState state) {
+        int maxConfiguredRequestSize = maxRequestSizeInBytes;
+        Optional<Integer> endpointMaxRequestSizeOverride = state.getEndpointForExecution().maxRequestSizeInBytesOverride();
```

Comment:
Never mind on moving stuff to longs. As you pointed out, we're ultimately limited by the fact that byte[], ByteBuf, etc lengths are ints. So without some major refactoring to allow for exposing payloads as InputStreams or something similar we're stuck with an int limit.